resolver/src/dns_lru.rs

```diff
@@ -63,7 +63,7 @@ impl DnsLru {
         let ttl_until = now + ttl;
 
         // insert into the LRU
-        let lookup = Lookup::new(Arc::new(rdatas));
+        let lookup = Lookup::new_with_ttl(Arc::new(rdatas), ttl_until);
```

Comment:
I suggest renaming things to be consistent. Your new API functions is called `valid_until`. I suggest renaming the `ttl_util` variable here the same thing. I also suggest renaming `Lookup::new_with_ttl` to `Lookup::new_valid_until`. Then it becomes easier to understand the code holistically.